### PR TITLE
Refactor code using a SearchProvider trait

### PR DESCRIPTION
**Pull request by ubruhin**
See issue #2

### CODE REVIEW

**Comment by rnestler** [comment]
Uh oh that will be hard to merge with #3 
Otherwise good job! Awesome :sparkles:

**Comment by rnestler** [comment]
We could merge #3 and rebase this onto it. If you don't mind?

**Comment by ubruhin** [comment]
ok, no problem

**Comment by rnestler** [comment]
The first commit is almost identical to the one already merged. When dropping it, rebasing should be relatively easy. Should I rebase it or do you want to do it?

**Comment by ubruhin** [comment]
done :)

File: src/search_providers/pirate_bay_search.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+        let mut res = self.connection.get(&url)
+            .header(Connection::close())
+            .send().unwrap();
+
+        let mut body = String::new();
+        res.read_to_string(&mut body).unwrap();
+        let document = Document::from_str(&body);
+
+        return parse_piratebay(&document);
```
Comment (rnestler) [inline src/search_providers/pirate_bay_search.rs]: One could replace this with just `parse_piratebay(&document)` without the `;`

File: src/search_providers/pirate_bay_search.rs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+use search_providers::SearchProvider;
+
+
+pub struct PirateBaySearch {
+    connection: hyper::Client
+}
+
+impl PirateBaySearch {
+    pub fn new() -> PirateBaySearch {
+        return PirateBaySearch{connection: Client::new()};
```
Comment (rnestler) [inline src/search_providers/pirate_bay_search.rs]: Same here: Just `PirateBaySearch{connection: Client::new()}`

**Comment by rnestler** [comment]
Looks good to me :+1: If you aren't working on this anymore I will merge it.

**Comment by ubruhin** [comment]
I stopped working on this, so it's ready to merge.

**Pull request merged**